stream/src/public_key/connection.rs

```diff
@@ -224,6 +225,7 @@ impl<St: Stream> Receiver<St> {
 #[cfg(test)]
 mod tests {
     use super::*;
+    use crate::{Receiver as _, Sender as _};
```

Comment:
It's not exactly clear to me why this is needed since the methods are defined in this file, however I guess the trait is defined in `mod.rs` so we still need to pull these into scope (?) The tests do not run without this line

Reply:
We don't import the traits explicitly in this file (using them as `crate::Sender`/`crate::Receiver`). Once we only make `send/receive` accessible behind these traits, we need to import said traits to call those functions.